src/Test.sol

```diff
@@ -139,6 +139,256 @@ abstract contract Test is DSTest {
             addr := create(0, add(bytecode, 0x20), mload(bytecode))
         }
     }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                    APPROX EQUAL
+    //////////////////////////////////////////////////////////////////////////*/
+
+    function assertApproxEq(
+        uint256 a,
+        uint256 b,
+        uint256 maxDelta
+    ) internal virtual {
+        uint256 delta = a > b ? a - b : b - a;
+
+        if (delta > maxDelta) {
+            emit log            ("Error: a ~= b not satisfied [uint]");
+            emit log_named_uint ("  Expected", b);
+            emit log_named_uint ("    Actual", a);
+            emit log_named_uint (" Max Delta", maxDelta);
+            emit log_named_uint ("     Delta", delta);
+            fail();
+        }
+    }
+
+    function assertApproxEq(
+        uint256 a,
+        uint256 b,
+        uint256 maxDelta,
+        string memory err
+    ) internal virtual {
+        uint256 delta = a > b ? a - b : b - a;
+
+        if (delta > maxDelta) {
+            emit log_named_string   ("Error", err);
+            emit log_named_uint     ("  Expected", b);
+            emit log_named_uint     ("    Actual", a);
+            emit log_named_uint     (" Max Delta", maxDelta);
+            emit log_named_uint     ("     Delta", delta);
+            fail();
+        }
+    }
+    function assertApproxEq(
```

Comment:
Seems you have the `assertApproxEq` methods duplicated?

Reply:
Four combinations:

- `uint256`
- `uint256` + user err message
- `int256`
- `int256` + user err message

Reply:
Ahh, yes missed that. We probably should add the `int256` versions of `assertApproxEqRel` also then, for consistency

Reply:
Done in 2c2337492baa475fa04bd3a539541d9d778a9c59